task01-anton11nahhas-main/tasks_week1_Anton_Nahhas/src/Vector.cpp

```diff
@@ -0,0 +1,272 @@
+		for (std::size_t i = 0; i < tempSize; i++) {
+			newArray[i] = m_array[i];
+		}
+		for (std::size_t i = tempSize; i < count; i++) {
+			newArray[i] = value;
+		}
+		delete[] m_array;
+		m_array = std::move(newArray);
+	}
+}
+
+//--------------------------------------------------------------------------------
+/*this function returns a pointer that points on the first cell of the vector, does not
+do anything if the vector is empty*/
+int* Vector::data() {
+	if (m_size != 0) {
+		return &m_array[0];
+	}
+}
+
+//--------------------------------------------------------------------------------
+/*this function clears the vector, meaning resizing the vector to 0 and emptying
+its values however it does not affect the capacity of the vector, the function 
+does not return anything*/
+void Vector::clear() {
+	m_size = 0;
+	m_array = 0;
+}
+
+//--------------------------------------------------------------------------------
+/*this function removes the last value in the vector, the function does not return
+anything*/
+void Vector::pop_back() {
+	if (m_size != 0) {
+		m_size--;
+	}
+}
+
+//--------------------------------------------------------------------------------
+/*this operator checks if the left hand side is equal to the right hand side, it 
+returns true if they are equal and false otherwise, it does so by checking first 
+if the sizes of both sides are equal and then check for each cell in both sides 
+if equal otherwise returns false*/
+bool operator==(const Vector& lhs, const Vector& rhs) {
+	if (lhs.getSize() == rhs.getSize()) {
+		for (int i = 0; i < lhs.getSize(); i++) {
+			if (lhs[i] != rhs[i]) {
+				return false;
+			}
+			return true;
+		}
+	}
+	return false;
+}
+
+//--------------------------------------------------------------------------------
+/*this function receives an iterator position and a valye, the value is inserted 
+before the position given, and returns the pointer that points on the position,
+it does so by looping until the position is found.*/
+int* Vector::insert(iterator pos, const int& val) {
+	int counter = 0,
+		index = 0;
+	iterator beg = begin();
+	std::size_t newSize = m_size + 1;
+	int* newArray = new int[m_size];
+	if (newSize >= m_capacity) {
+		resize(newSize);
+	}
+	else {
+		m_size = newSize;
+	}
+
+	while (index < m_size) {
+		if (beg != pos) {
+			newArray[index] = m_array[counter];
+			counter++;
+		}
+		else {
+			newArray[index] = val;
+		}
+		index++;
+		++beg;
+	}
+	delete[] m_array;
+	m_array = std::move(newArray);
+
+
+	return pos;
+
+}
+
+//--------------------------------------------------------------------------------
+/*this function receives an iterator, and its goal to delete the element on that 
+position that the iterator is pointing at, returns the pointer that the pos is
+pointing at. It does so by searching for the position required and deleting it*/
+int* Vector::erase(iterator pos) {
+	iterator beg = begin();
+	int* newArray = new int[m_size--];
```

Comment:
should it be? int* newArray = new int[--m_size];. or m_size-- ?  https://stackoverflow.com/questions/24853/what-is-the-difference-between-i-and-i